sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -1150,3 +1150,53 @@ def test_gradient_boosting_early_stopping():
 
     assert gbc.n_estimators_ == 100
     assert gbr.n_estimators_ == 200
+
+
+def test_gradient_boosting_validation_fraction():
+    X, y = make_classification(n_samples=1000, random_state=0)
+
+    gbc = GradientBoostingClassifier(n_estimators=100,
```

Comment:
(Instead of looping or using pytests's parametrize, I used clone to improve readability.)